### DISCUSSION TITLE
Recover deleted transactions?

### DISCUSSION THREAD

**Discussion by isaacgr** [firefly-iii/firefly-iii#6928] [Firefly III]
I notice that, by design, according to other posts, the database keeps deleted transactions. This is causing an issue for a script I'm writing as subsequent POST requests of the same information still exist and trigger a 'duplicate' error when I have 'error_if_duplicate_hash' set to true (which is fine moving forward, but is something I'd like to clear up). 

The database has a 'deleted_at' field in the transactions table. Should I just make this NULL to restore them?

**Comment by JC5** [comment]
Yes, that should work.

**Comment by elChapoSing** [comment]
Hello - 
First great work on the tool. I'm just getting started so obviously I deleted some transactions I did not want to delete :)

Unfortunately, this solution does not do the trick for me.

I did
> select * from transactions where deleted_at is not null;

Which returned 8 lines

Then
> update transactions set deleted_at = NULL  where deleted_at is not null;

When I rerun the select, it returns no records as expected.

Yet the transactions I wanted to restore are not there.

Do I need to trigger a refresh of some sort from the UI ?
@isaacgr did it just work immediately for you ?

**Comment by elChapoSing** [comment]
Update - I had to delete the transactions from the `transaction_journals` table and reimport for it to work.
Could I have done that from the get go @JC5 ?

**Reply by JC5** [reply]
No, the tables are a bit more extended nowadays: undelete from transactions, transaction_journals AND transaction_groups.

**Reply by elChapoSing** [reply]
Thks for the quick reply !
ok I have now reimported stuff so will see if it needs cleaning up...

**Reply by elChapoSing** [reply]
They were indeed still in `transaction_journals` - will add my voice to the "manage deleted transaction page" feature request.
